staff/sean-daryanani/pre-course/calculadora/main.js

```diff
@@ -0,0 +1,313 @@
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//Declare DOM variables
+var numbers = document.querySelectorAll(".num");
+var operators = document.querySelectorAll(".operator");
+var input = document.querySelector("input");
+var clear = document.querySelector(".clear");
+var plusMinus = document.querySelector(".plusminus");
+var percent = document.querySelector(".percent");
+var equals = document.querySelector(".equals");
+var decimal = document.querySelector("#dec");
+var squareRoot = document.querySelector(".sqrt")
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//Declare global variables
+var firstNumber=0;
+var secondNumber=0;
+var currentOperand;
+var result = 0;
+var currentNumber = 0;
+var operandSelected;
+var calcStatus= {
+    first:true,
+    second:false,
+    result:false,
+    notNumberClicked:false,
+};
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//Functions
+
+
+function removeBackground() {
+    for (var i=0; i<numbers.length; i++) {
+        numbers[i].classList.remove("selected");
+        if (operators[i]) {
+            operators[i].classList.remove("selected");
+        }
+    }
+}
+
+
+function makeVisible() {
+    for (var i=0 ; i<operators.length; i++) {
+        operators[i].style.visibility="visible";
+    }
+    plusMinus.style.visibility = 'visible';
+    percent.style.visibility = 'visible';
+
+}
+
+function hideOperators() {
+    if (currentOperand === '+') {
+        operators[0].style.visibility= 'hidden';
+        operators[1].style.visibility= 'hidden';
+        operators[2].style.visibility= 'hidden';
+        plusMinus.style.visibility = 'hidden';
+        percent.style.visibility = 'hidden'

```

Comment:
as far as I understand, the use of  semicolon(;) is optional... but I see a pattern here that confuses me: why do you always omit only the last semicolon?

Reply:
I prefer to put semicolons "por si acaso". I really don't know why I missed it out in the last one always... no logic behind that! Thanks

Reply:
I added a semicolon to all the lines now